tests/unit/generator/test_options.py

```diff
@@ -161,3 +159,18 @@ def test_options_bool_flags():
 
         options = Options.build(opt)
         assert getattr(options, attr)
+
+    # Check autogen-snippets separately, as it is default True
+    options = Options.build("")
+    assert options.autogen_snippets
+
+    options = Options.build("autogen-snippets=False")
+    assert not options.autogen_snippets
+
+
+def test_options_autogen_snippets_false_for_old_naming():
+    # NOTE: Snippets are not currently correct for the alternative (Ads) templates
+    # so always disable snippetgen in that case
+    # https://github.com/googleapis/gapic-generator-python/issues/1052
+    options = Options.build("old-naming")
+    assert not options.autogen_snippets
```

Comment:
I think, just for the sake of completeness, it would be good to have a test where both `autogen-snippets=True` and `old-naming` are set to make sure that the builder logic handles that correctly.